modules/NuGetPackageVerifier/console/Rules/PackageSigningrule.cs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using Microsoft.Extensions.CommandLineUtils;
+
+namespace NuGetPackageVerifier.Rules
+{
+    public class PackageSigningRule : IPackageVerifierRule
+    {
+        private readonly string _nuGetExePath;
+
+        public PackageSigningRule()
+            : this(GetKorebuildNuGetPath())
+        {
+        }
+
+        public PackageSigningRule(string exePath)
+        {
+            if (!File.Exists(exePath))
+            {
+                throw new ArgumentException($"NuGet.exe could not be located at {exePath}", nameof(exePath));
+            }
+
+            _nuGetExePath = exePath;
+        }
+
+        private static string GetKorebuildNuGetPath()
+        {
+            // KoreBuild/nuget.exe
+            // KoreBuild/modules/NuGetPackageVerifier/
+            return Path.Combine(AppContext.BaseDirectory, "..", "..", "nuget.exe");
+        }
+
+        public IEnumerable<PackageVerifierIssue> Validate(PackageAnalysisContext context)
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                throw new InvalidOperationException("Package sign verification is only supported on Windows machines");
+            }
+
+            var args = new[]
```

Comment:
Adding tests turned out to be useful. The sign verification only works on desktop \ Windows: https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Packaging/PackageArchiveReader.cs#L268-L282. It's just easier to shell out to the next than trying to cross-compile the console app. We already require Windows for authenticode verification, so this is yet another thing that requires it.